core/src/main/java/org/elasticsearch/transport/ConnectionProfile.java

```diff
@@ -38,13 +39,22 @@
      * A pre-built light connection profile that shares a single connection across all
      * types.
      */
-    public static final ConnectionProfile LIGHT_PROFILE = new ConnectionProfile(
-        Collections.singletonList(new ConnectionTypeHandle(0, 1, EnumSet.of(
-            TransportRequestOptions.Type.BULK,
-            TransportRequestOptions.Type.PING,
-            TransportRequestOptions.Type.RECOVERY,
-            TransportRequestOptions.Type.REG,
-            TransportRequestOptions.Type.STATE))), 1, null, null);
+    public static final ConnectionProfile LIGHT_PROFILE = getLightProfileWithTimeout(null, null);
+
+    /**
+     * Builds a light connection profile that shares a single connection across all
+     * types, using the given timeouts
+     */
+    public static ConnectionProfile getLightProfileWithTimeout(@Nullable TimeValue connectTimeout,
+                                                               @Nullable TimeValue handshakeTimeout) {
+        return new ConnectionProfile(
```

Comment:
I wonder if we should do this. I think we should move the `LIGHT_PROFILE` into tests somewhere and then require every special use to build it's own. The problem I have here is that the `getLightProfileWithTimeout` shares one connection across all uses. I think in the case of ping we should only use 1 connection for `PING` and 0 for the others. that will cause an exception if it's used in a wrong context. makes sense?